### PR TITLE
Put k8schain in its own go module

### PR DESCRIPTION
**Pull request by jonjohnsonjr**
This should make go-containerregistry much easier to use.

https://github.com/google/go-containerregistry/pull/902
https://github.com/google/ko/issues/287
https://github.com/google/ko/issues/258#issuecomment-756243939

WANT_LGTM=all

### CODE REVIEW

**Review by ImJasonH — APPROVED** [review]

**Comment by jonjohnsonjr** [comment]
Per https://pkg.go.dev/github.com/google/go-containerregistry/pkg/authn/k8schain?tab=importedby y'all might be affected.

cc @andreyvelich 
cc @scothis
cc @matthewmcnew 
cc @hasheddan 
cc @davidewatson (also, https://github.com/davidewatson/keychain looks interesting, would love to see a README 😉)
cc @leogr

**Comment by jonjohnsonjr** [comment]
Split this up into two commits:

1. go.mod changes, should be easy to review
2. go mod tidy && go mod vendor

I didn't vendor the k8schain deps because I think it might break some expectations of various tools (only one top-level vendor/ directory). That's why there's the huge negative diff. Hopefully that's fine.

**Comment by mattmoor** [comment]
Did you add any presubmit safeguard that the nested module files are correct?

**Comment by jonjohnsonjr** [comment]
cc @tejal29 as well

**Comment by jonjohnsonjr** [comment]
> Did you add any presubmit safeguard that the nested module files are correct?

Done: https://github.com/google/go-containerregistry/pull/904/commits/52dc010d199eacf52ea60176701af16cc180bc33

**Review by hasheddan — APPROVED** [review]
Thanks @jonjohnsonjr! I'll test this out with our @crossplane usage :+1:

**Comment by andreyvelich** [comment]
Thank you for the notice @jonjohnsonjr!
cc @gaocegege @johnugeorge

File: pkg/authn/k8schain/go.mod [hunk 1]
```diff
@@ -0,0 +1,23 @@
+module github.com/google/go-containerregistry/pkg/authn/k8schain
+
+go 1.15
```
Comment (mattmoor) [inline pkg/authn/k8schain/go.mod:3]: the parent is 1.14, these should be consistent.

Comment (jonjohnsonjr) [reply pkg/authn/k8schain/go.mod]: I think it would be fine for them to diverge in the future, but I've dropped this to 1.15.

**Pull request merged**